cli/cmd/upgrade.go

```diff
@@ -0,0 +1,95 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Brains-Beyond-Expectations/bbe-quest/interfaces"
+	"github.com/Brains-Beyond-Expectations/bbe-quest/misc/logger"
+	"github.com/Brains-Beyond-Expectations/bbe-quest/services/config_service"
+	"github.com/Brains-Beyond-Expectations/bbe-quest/services/helper_service"
+	"github.com/Brains-Beyond-Expectations/bbe-quest/services/package_service"
+	"github.com/Brains-Beyond-Expectations/bbe-quest/services/ui_service"
+	"github.com/spf13/cobra"
+)
+
+var upgradeCmd = &cobra.Command{
+	Use:     "upgrade",
+	Aliases: []string{"u"},
+	Short:   "Upgrade BBE packages",
+	Args:    cobra.ExactArgs(0),
+	Run: func(cmd *cobra.Command, args []string) { // coverage-ignore
+		helperService := helper_service.HelperService{}
+		uiService := ui_service.UiService{}
+		configService := config_service.ConfigService{}
+		packageService := package_service.PackageService{}
+
+		uninteractive, _ := cmd.Flags().GetBool("yes")
+
+		err := upgradeCommand(helperService, uiService, configService, packageService, uninteractive)
+		if err != nil {
+			logger.Error("", err)
+			os.Exit(1)
+		}
+	},
+}
+
+func upgradeCommand(helperService interfaces.HelperServiceInterface, uiService interfaces.UiServiceInterface, configService interfaces.ConfigServiceInterface, packageService interfaces.PackageServiceInterface, uninteractive bool) error {
+	bbeConfig, err := configService.GetBbeConfig(helperService)
```

Comment:
Shall we have this pre check as a separated function? We gonna be doing the same validation a fair bit I think

Reply:
Actually I don't think it makes sense, we're not gonna get around having to handle this case specifically anyway. I would not make this change 🤷